### PR TITLE
Fix gprofiler2 genesets default object

### PR DESCRIPTION
**Pull request by WackerO**
Added empty channel for gprofiler2 when no gene sets are provided

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/differentialabundance.nf [hunk 1]
```diff
@@ -73,6 +73,8 @@ if (run_gene_set_analysis) {
     } else if (params.gprofiler2_run) {
         if (!params.gprofiler2_token && !params.gprofiler2_organism) {
             error("To run gprofiler2, please provide a run token, GMT file or organism!")
+        } else {
+            ch_gene_sets = [[]]     // If one of token or organism is provided, make gene_sets empty (nested because of .first() in the gprofiler call)
```
Comment (pinin4fjords) [inline workflows/differentialabundance.nf]:
```suggestion
            ch_gene_sets = Channel.of([])
```

... for consistency with:

```
 ch_gene_sets = Channel.of(gene_sets_files).map { file(it, checkIfExists: true) }
```

Also, the logic for this block would be simpler if we just did this straight after:

```
if (run_gene_set_analysis) {
```

Then we could remove the last `else`.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Looks good, thanks for chasing, just think we can simplify

File: workflows/differentialabundance.nf (hunk 1 above)
Comment (WackerO) [reply workflows/differentialabundance.nf]: Ah of course!

**Review by pinin4fjords — APPROVED** [review]
LGTM, thanks for chasing this down

**Pull request merged**